backend/core/src/migration/1630250097191-adds-jurisdiction-relations.ts

```diff
@@ -0,0 +1,71 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+
+export class addsJurisdictionRelations1630250097191 implements MigrationInterface {
+    name = 'addsJurisdictionRelations1630250097191'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`DROP INDEX "IDX_8317da96d5a775889e2631cc25"`);
+        // get translation county codes before rename
+        const translations = await queryRunner.query(`SELECT id, county_code FROM translations`)
+        await queryRunner.query(`ALTER TABLE "translations" RENAME COLUMN "county_code" TO "jurisdiction_id"`);
+        await queryRunner.query(`CREATE TABLE "user_accounts_jurisdictions_jurisdictions" ("user_accounts_id" uuid NOT NULL, "jurisdictions_id" uuid NOT NULL, CONSTRAINT "PK_66ae1ae446619b775cafb03ce4a" PRIMARY KEY ("user_accounts_id", "jurisdictions_id"))`);
+        await queryRunner.query(`CREATE INDEX "IDX_e51e812700e143101aeaabbccc" ON "user_accounts_jurisdictions_jurisdictions" ("user_accounts_id") `);
+        await queryRunner.query(`CREATE INDEX "IDX_fe359f4430f9e0e7b278e03f0f" ON "user_accounts_jurisdictions_jurisdictions" ("jurisdictions_id") `);
+        // assign jurisdiciton ID from county code
+        await queryRunner.query(`
+          UPDATE listings
+          SET jurisdiction_id = j.id
+          FROM listings AS l
+            INNER JOIN jurisdictions AS j
+              ON l.county_code = j.name
+        `)
+        // drops county code from listings
+        await queryRunner.query(`ALTER TABLE "listings" DROP COLUMN "county_code"`);
+        await queryRunner.query(`ALTER TABLE "jurisdictions" ADD CONSTRAINT "UQ_60b3294568b273d896687dea59f" UNIQUE ("name")`);
+        await queryRunner.query(`ALTER TABLE "translations" DROP COLUMN "jurisdiction_id"`);
```

Comment:
`translations.county_code` has been first renamed to `jurisdiction_id` and here it's dropped. Isn't it a data loss?

Reply:
I think the first rename should be county_code to `name` instead of `jurisdiction_id` considering what this UPDATE is doing:

```
        await queryRunner.query(`
          UPDATE listings
          SET jurisdiction_id = j.id
          FROM listings AS l
            INNER JOIN jurisdictions AS j
              ON l.county_code = j.name
        `)
```

Reply:
@pbn4 ,

> `translations.county_code` has been first renamed to `jurisdiction_id` and here it's dropped. Isn't it a data loss?

No, because before this it fetches the translations so it can set them based off of the county_code before it was dropped.



> I think the first rename should be county_code to `name` instead of `jurisdiction_id` considering what this UPDATE is doing:
> 
> ```
>         await queryRunner.query(`
>           UPDATE listings
>           SET jurisdiction_id = j.id
>           FROM listings AS l
>             INNER JOIN jurisdictions AS j
>               ON l.county_code = j.name
>         `)
> ```

This is for listings though, not translations, which has the rename (listings doesn't), and this occurs before county_code is dropped from listings.